Mathlib/Algebra/MonoidAlgebra/Basic.lean

```diff
@@ -162,16 +162,18 @@ section Mul
 
 variable [Semiring k] [Mul G]
 
+irreducible_def mul' (f g : MonoidAlgebra k G) : MonoidAlgebra k G :=
```

Comment:
Do we get the same benefits with just `@[irreducible]`?

Reply:
I didn't check. Are there advantages of `@[irreducible]`?

Reply:
Yeah you get the same benefits in terms of the speed-up. 

The auto-`*_def` declarations are convenient but you can force things with `with_unfolding_all` if necessary. 

I would generally prefer to depend on upstream tooling unless we absolutely need something that can/will not be provided. But I don't feel strongly enough to put up a fight.

Reply:
Those are good points. 
We get the same improvement, with `@[irreducible]`, so I modified this PR.

Reply:
Using the `@[irreducible]` is there a way to "unfold irreducible definition `X`, but no other irreducible definitions"? I tried `simp [mul']` or `unfold mul'`, but they both seem to fail.

Reply:
I don't know of a way to that unfortunately.

Reply:
We also used `private` for `Real`.

Reply:
That is orthogonal to irreducibility, and I don't particularly care about making this private.

Reply:
I would like to have a consistent design across the library, or there is a reason why the design is different. But I would not insist.

Reply:
Better to do things like this (improving consistency) in separate PRs, rather than bundling it in because the relevant file was touched.